### ISSUE TITLE
Registration issue

### ISSUE THREAD

**Issue by Kachuk** [tgalal/yowsup#2152]
im having this issue when i try to register a number

yowsup-cli v2.0.15
Using yowsup v2.5.2

INFO:yowsup.common.http.warequest:{"login":"54XXXXXXXX","status":"fail","reason":"old_version"}

status: fail
reason: old_version
login: 54XXXXXXXXXX
 i have tried the solutions in other posts but none works

**Comment by joko3ana** [comment]
see 
https://github.com/tgalal/yowsup/issues/2090

**Comment by plehr** [comment]
same here:
`
root@vpsxxxx:~/yowsup# ./yowsup-cli registration --requestcode sms --phone 4917xxxxxxxx --cc 49 --mcc 2xx --mnc 0x
yowsup-cli  v2.0.15
yowsup      v2.5.2

Copyright (c) 2012-2016 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


INFO:yowsup.common.http.warequest:{"login":"4917xxxxxxxx ","status":"fail","reason":"no_routes","retry_after":3600,"sms_wait":0,"voice_wait":0}

status: fail
retry_after: 3600
reason: no_routes
login: 4917xxxxxxxx 
`
yowsup/env/env_android.py:`
 _MD5_CLASSES = "1naz8gL5pIYWbtaOZ3207g=="
_VERSION = "2.17.16"
_OS_NAME = "Android"`
yowsup/env/env.py: `DEFAULT = "android"`

**Comment by Alejandro94elx** [comment]
Hey: 

Change in yowsup/env/env_android.py


Works NOW:
  _MD5_CLASSES = "ox998VW0nBTueMVfjuZkmQ=="
  _VERSION = "2.17.212"

**Comment by Kachuk** [comment]
none of these works, im getting the same error

**Comment by coolnickname** [comment]
I noticed this too. A few days ago, around the same time this issue was made, Whatsapp updated. I've tried to update my Android enviroment with:
`_VERSION = "2.17.251"`
`_MD5_CLASSES = "KoK/8Dy+k1yOP80RObNznw=="`
With no success. I've even tried to extract the enviroment data from my old Galaxy Fame, but I always get old_version.

Something to note is that a few lines before
`INFO:yowsup.common.http.warequest:{"login":"redacted","status":"fail","reason":"old_version"}`
it says
`INFO:yowsup.common.http.warequest:{"login":"redacted","status":"fail","reason":"incorrect","sms_length":6,"voice_length":6,"sms_wait":0,"voice_wait":0}`
I'm not sure if that's normal.

**Comment by coolnickname** [comment]
On a side note, when I try to login with fake data I get the infamous

> not-authorized

I know most of the times this would mean I've been banned, IP banned at least, since my number still works. But could this be caused by the fake data too?

In the mean time I might as well go search for some good proxies. Can anyone recommend a good, preferably free, proxy service?

**Comment by weberxw** [comment]
this old_version I resolved simple executing `./yowsup-cli ` instead `yowsup-cli` at the registration time...... but after that i'm getting not-authorized when I try to send a sms :/

**Comment by nurdan** [comment]
Add/Use -E Android option for register command line

**Comment by coolnickname** [comment]
I'm using my Android enviroment for registering, and I'm not sure how it would help but I've also executed yowsup-cli from the directory.

**Comment by coolnickname** [comment]
Update: I used a proxy, but it still didn't work. But after some fiddling around I ended up using the binary from the folder (./yowsup-cli) as the root user, which I didn't do before, and it worked! 

I used Whatsapp version 2.17.251, for which I posted the MD5 above.

I don't think the proxy really helped since the first attempt with the proxy did the same as without, but now I'll just keep using it for a while since Whatsapp might get suspicious about my fast traveling.
